src/pages/index.js

```diff
@@ -29,54 +31,195 @@ const enableValidation = (config) => {
 newCardPopup.setEventListeners();
 
+export const deleteConfirmPopup = new PopupConfirmation(
+  "#modal-delete",
+  handleDeleteConfirmation
+);
+deleteConfirmPopup.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#modal-avatar", handleUpdateAvatar);
+avatarPopup.setEventListeners();
+
+//HANDLES
+
+// function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+//   popupInstance.renderLoading(true, loadingText);
+//   request()
+//     .then(() => {
+//       popupInstance.close();
+//     })
+//     .catch(console.error)
+//     .finally(() => {
+//       popupInstance.renderLoading(false);
+//     });
+// }
+
+// function handleProfileFormSubmit(inputValues) {
+//         we create a function that returns a promise
+//   function makeRequest() {
+//        `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
+//     return api.editProfile(inputValues).then((userData) => {
+//       userInfo.setUserInfo(userData)
+//     });
+//   }
+//         Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
+//   handleSubmit(makeRequest, profilePopup);
+// }
+
+function handleDeleteConfirmation(card, id) {
+  api
+    .deleteCard(id)
+    .then(() => {
+      card.handleDeleteCard();
+      deleteConfirmPopup.close();
+    })
+    .catch((err) => console.error(`${err}, Failed to delete card`));
+}
+
+function handleProfileUpdate(userData) {
+  profilePopup.setLoading(true);
+  api
+    .updateUserData(userData)
+    .then(() => {
+      initialUserData.setUserInfo(userData);
+      profilePopup.close();
+    })
+    .catch((err) => console.error(`${err}, Failed to update Profile Info`))
+    .finally(() => {
+      profilePopup.setLoading(false);
+      profilePopup.setSave();
+    });
+}
+
+function handleAddCard(cardData) {
+  newCardPopup.setLoading(true);
+  api
+    .createCard(cardData)
+    .then((data) => {
+      const cardFormElement = createCard(data);
+      cardSection.addItem(cardFormElement);
+      newCardPopup.close();
+    })
+    .catch((err) => console.error(`${err}, Failed to add Card`))
+    .finally(() => {
+      newCardPopup.setLoading(false);
+      newCardPopup.setSave();
+    });
+}
+
+function handleUpdateAvatar(input) {
+  avatarPopup.setLoading(true);
+  api
+    .updateAvatar(input.link)
+    .then(() => {
+      initialUserData.setUserAvatar(input.link);
+      avatarPopup.close();
+    })
+    .catch((err) => console.error(`${err}, Failed to update Avatar`))
+    .finally(() => {
+      avatarPopup.setLoading(false);
+      avatarPopup.setSave();
+    });
+}
+
+//EVENT LISTENER
+
 profileEditButton.addEventListener("click", () => {
   formValidators["profileForm"].resetValidation();
-  const data = userInfo.getUserInfo();
-  profileNameInput.value = data.name;
-  profileJobInput.value = data.job;
-  profilePopup.open();
+  api
+    .getUserData()
```

Comment:
Please, don't call `api.getUserData()` on profile opening because it's a waste of resources. Please, use method  `getUserInfo` from class `UserInfo`


* You need to have method `getUserInfo` inside `UserInfo` according to the task of the 8th sprint (the same is true for the 9th one)

Reply:
fixed this issue